SAKPaaS/src/app/core/interceptors/api.interceptor.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable, Provider, forwardRef } from '@angular/core';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent, HTTP_INTERCEPTORS } from '@angular/common/http';
+import { Observable, ReplaySubject } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { AuthKeycloakService } from '../services/auth-keycloak.service';
+
+export const API_INTERCEPTOR_PROVIDER: Provider = {
+  provide: HTTP_INTERCEPTORS,
+  useExisting: forwardRef(() => ApiInterceptor),
+  multi: true
+};
+
+@Injectable()
+export class ApiInterceptor implements HttpInterceptor {
+  private token$ = new ReplaySubject<string>(1);
+
+  constructor(private authService: AuthKeycloakService) {
+    authService.getToken().subscribe(this.token$);
+  }
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    // Apply the headers
+    return this.token$.pipe(switchMap(token => {
```

Comment:
Wenn man nicht angemeldet ist und keinen Token hat, würde das hier dann nicht alle API-Requests blockieren? Das ReplaySubject emitted doch erst nachdem es überhaupt einen Wert bekommen hat, oder?

Reply:
Ähm no eigentlich nicht, weil der `$token` im `AuthService` ein BehaviorSubject mit dem Defaultwert `null` ist.

Ich weiß gar nicht, warum das im `Interceptor` ein ReplaySubject ist @timgrohmann? :D

Reply:
Weil wir das so brauchten dass es auf das Token wartet, aber du kannst da wahrscheinlich einfach ein `Observable` draus machen.

Reply:
habs grad nochmal getestet, geht auch, wenn man nicht eingeloggt is :heavy_check_mark:

Reply:
habs geändert :)